### PR TITLE
Completely refactor `launch()`

### PR DESCRIPTION
**Pull request by CommandMC**
With the recent addition of GOG support, launching games became a bit confusing. Now, the general "launch" function had to not only do different things depending on the user's and game's platform, but also had to differentiate between Epic and GOG. This resulted in it becoming very hard to read/understand  
This PR completely restructures this functionality by splitting the function up into smaller parts and making these parts available as functions, then calling the different functions in LegendaryGame/GOGGame's own `launch` functions.

It also brings a lot of small improvements:
- Game logs are no longer weirdly indented
- The Steam Runtime is no longer exclusive to native GOG games (still needs frontend update)
- `prepareLaunch` (new) now checks for both MangoHud and GameMode if they're enabled
- With `prepareWineLaunch` and `verifyWinePrefix` (both new), Wine/CX & Proton prefixes are now created properly & the GOG setup is only ran when necessary
- VKD3D is also auto-installed on game launch if it's enabled

Currently this is feature-complete, but I still need to test on Windows, Mac and a clean Linux install

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [X] Tested the feature and it's working on a current and clean install.
- [X] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)

### CODE REVIEW

**Comment by CommandMC** [comment]
Alright, this seems to work fine on Windows. Can't test on Mac sadly

**Comment by CommandMC** [comment]
Note: Currently this has merge conflicts, these will be outdated once #1092 is merged so I'll wait on that for now

**Review by flavioislima — APPROVED** [review]
A pretty good refactor and everything seems to be working fine!
Tested GOG and Epic natives and non-natives games on a Mac and they are being launched.
Just need to update the branch and you are good to go IMO.
@imLinguin could take a look at it as well if you have the time.

**Comment by flavioislima** [comment]
@CommandMC merged the other one. We can merge this one after fixing the conflicts 👍🏽

File: electron/launcher.ts
Comment (flavioislima) [inline electron/launcher.ts:212]: what `ret` means? 👀

**Review by flavioislima — DISMISSED** [review]
Looks pretty good to me.
Well documented and easy to understand. ⚔️

File: electron/launcher.ts
Comment (CommandMC) [reply electron/launcher.ts:212]: Short for `return`, so it's the object that's returned by the function  
Very creative, I know

**Review by imLinguin — DISMISSED** [review]
Great PR

**Review by flavioislima — APPROVED** [review]
Tested here:
- Linux: Both GOG and Epic games working fine. both Windows and native. Installing and other stuff as well working good.
- Mac: Same, both GOG and Epic, both native and windows games are working fine. Tested also: login, install, uninstall, just in case, and its working fine
- Windows: Both Epic and GOG games are installing and launching just fine as well 🎉

**Pull request merged**
